process/block/shardblock.go

```diff
@@ -857,7 +851,11 @@ func (sp *shardProcessor) CommitBlock(
 	processedMiniBlocks := process.ConvertProcessedMiniBlocksMapToSlice(sp.processedMiniBlocks)
 	sp.mutProcessedMiniBlocks.RUnlock()
 
-	sp.prepareDataForBootStorer(headerInfo, header.Round, finalHeaders, finalHeadersHashes, processedMiniBlocks)
+	if len(selfNotarizedHeaders) > 0 {
+		sp.lowestNonceInSelfNotarizedHeaders = selfNotarizedHeaders[0].GetNonce()
```

Comment:
We need it, because not each round selfNotarizedHeaders variable contains something (depending if metachain headers processed contains or not some info about your shard). And we need to set in bootstorer for each shard header committed the lowest final block nonce determined from the sp.getHighestHdrForOwnShardFromMetachain